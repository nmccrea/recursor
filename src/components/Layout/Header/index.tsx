import React from "react"
import { useStaticQuery, graphql } from "gatsby"
import { Panel } from "components/parts/Panel"
import { HiddenContainer, H1, GithubImage } from "./styled"

const Header = () => {
  const { site } = useStaticQuery(
    graphql`
      query {
        site {
          siteMetadata {
            title
            githubUrl
          }
        }
      }
    `
  )

  return (
    <HiddenContainer>
      <Panel>
        <H1>{site.siteMetadata.title}</H1>
      </Panel>

      <a href={site.siteMetadata.githubUrl} target="blank">
        <GithubImage />
      </a>
    </HiddenContainer>
  )
}

export default Header
