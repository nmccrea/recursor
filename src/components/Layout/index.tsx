import React from "react"
import GlobalStyle from "../styles/GlobalStyle"

interface Props {
  children: React.ReactNode
}

const Layout = ({ children }: Props) => (
  <>
    <GlobalStyle />
    {children}
  </>
)

export default Layout
