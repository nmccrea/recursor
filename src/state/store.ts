/**
 * Configures and exports the redux store and other top-level Redux resources.
 * @packageDocumentation
 */

import { configureStore, combineReducers } from "@reduxjs/toolkit"
import similarities from "./similarities/reducer"

const rootReducer = combineReducers({ similarities })
const createStore = () => configureStore({ reducer: rootReducer })

export { createStore }
export type RootState = ReturnType<typeof rootReducer>
export type Selector<ReturnType> = (state: RootState) => ReturnType | undefined
