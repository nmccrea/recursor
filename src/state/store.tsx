/**
 * Configures and exports the redux store and other top-level Redux resources.
 * @packageDocumentation
 */

import { combineReducers, configureStore } from "@reduxjs/toolkit"

const rootReducer = combineReducers({})
export type RootState = ReturnType<typeof rootReducer>
export default configureStore({ reducer: rootReducer })
