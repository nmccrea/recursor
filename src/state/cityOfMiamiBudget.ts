/**
 * TODO: type
 * TODO: test
 * TODO: doc
 */

import { RootState } from "./store"
import {
  createAsyncThunk,
  createSlice,
  CaseReducer,
  PayloadAction,
} from "@reduxjs/toolkit"

// State
enum Status {
  Idle = "idle",
  Pending = "pending",
  Fulfilled = "fulfilled",
}

interface CityOfMiamiBudgetState {
  status: Status
  data?: object
}

const INITIAL_STATE: CityOfMiamiBudgetState = { status: Status.Idle }

// Case Reducers
const fetchDataPending: CaseReducer<CityOfMiamiBudgetState, PayloadAction> = (
  previousState
) => ({
  ...previousState,
  status: Status.Pending,
})

const fetchDataFulfilled: CaseReducer<
  CityOfMiamiBudgetState,
  PayloadAction<object>
> = (previousState, action) => ({
  ...previousState,
  status: Status.Fulfilled,
  data: action.payload,
})

// Async Thunks
const fetchData = createAsyncThunk("fetchData", async (url: string) => {
  const response = await fetch(url)
  return await response.json()
})

// Slice
const { reducer } = createSlice({
  name: "cityOfMiamiBudget",
  initialState: INITIAL_STATE,
  reducers: {},
  extraReducers: {
    [fetchData.pending.type]: fetchDataPending,
    [fetchData.fulfilled.type]: fetchDataFulfilled,
  },
})

// Selectors
const getStatus = (state: RootState) =>
  (state.cityOfMiamiBudget && state.cityOfMiamiBudget.status) || undefined

const getData = (state: RootState) =>
  (state.cityOfMiamiBudget && state.cityOfMiamiBudget.data) || undefined

export default reducer
export { fetchData, getStatus, getData, Status }
