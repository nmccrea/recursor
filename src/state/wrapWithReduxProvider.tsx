/**
 * Implements Gatsby's `wrapRootElement` API.
 * @packageDocumentation
 */

import React from "react"
import { Provider as ReduxProvider } from "react-redux"
import store from "./store"

interface Arg {
  element: React.ReactNode
}
export const wrapWithReduxProvider = ({ element }: Arg) => (
  <ReduxProvider store={store}>{element}</ReduxProvider>
)
