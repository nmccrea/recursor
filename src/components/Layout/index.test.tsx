import React from "react"
import { Provider } from "react-redux"
import { render } from "@testing-library/react"
import { createStore } from "../../state/store"
import Layout from "."

describe("`<Layout />`", () => {
  it("renders correctly", () => {
    const { container } = render(
      <Provider store={createStore()}>
        <Layout>
          <p>Test Content</p>
        </Layout>
      </Provider>
    )

    expect(container).toMatchSnapshot()
  })
})
