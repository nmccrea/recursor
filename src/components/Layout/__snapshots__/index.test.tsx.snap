// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`<Layout />\` renders correctly 1`] = `
.c1 {
  padding: 1rem;
  background-color: rgba(255,255,255,0.7);
  border-radius: 0.5rem;
  box-shadow: 0px 3px 1px -2px rgba(0,0,0,0.2), 0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12);
}

.c0 {
  position: fixed;
  top: 1rem;
  left: 0;
  width: 100vw;
  box-sizing: border-box;
  margin: 0;
  padding: 0 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  visibility: hidden;
}

.c0 > * {
  visibility: visible;
}

.c2 {
  font-family: "Audiowide",system-ui,-apple-system,BlinkMacSystemFont, "Segoe UI",Roboto,"Helvetica Neue",Arial,"Noto Sans",sans-serif, "Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol","Noto Color Emoji";
  color: cornflowerblue;
  text-transform: uppercase;
  line-height: 1;
  margin: 0;
}

.c3 {
  background-image: url(-- MOCK FILE EXPORT VALUE --);
  background-size: 1.6rem 1.6rem;
  width: 1.6rem;
  height: 1.6rem;
  opacity: 0.6;
  margin: 0 1rem;
}

.c3:hover {
  opacity: 0.8;
}

<div>
  <p>
    Test Content
  </p>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <h1
        class="c2"
      >
        Mock Site Name
      </h1>
    </div>
    <a
      href="https://github.com/mockiemcgee/test"
      target="blank"
    >
      <div
        class="c3"
      />
    </a>
  </div>
</div>
`;
